engine/class_modules/apl/apl_monk.cpp

```diff
@@ -648,51 +654,99 @@ namespace monk_apl
+    heavy_aoe->add_action( "whirling_dragon_punch,if=active_enemies>8" );
+    heavy_aoe->add_action( "fists_of_fury" );
+    heavy_aoe->add_action( "spinning_crane_kick,if=buff.bonedust_brew.up&combo_strike" );
+    heavy_aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.stack=3&talent.shadowboxing_treads" );
+    heavy_aoe->add_action( "whirling_dragon_punch,if=active_enemies>=5" );
+    heavy_aoe->add_action( "rising_sun_kick,if=min:debuff.mark_of_the_crane.remains,if=talent.whirling_dragon_punch&cooldown.whirling_dragon_punch.remains<3&cooldown.fists_of_fury.remains>3&!buff.kicks_of_flowing_momentum.up" );
+    heavy_aoe->add_action( "spinning_crane_kick,if=min:debuff.mark_of_the_crane.remains,if=combo_strike&cooldown.fists_of_fury.remains<5&buff.chi_energy.stack>10" );
+    heavy_aoe->add_action( "spinning_crane_kick,target_if=min:debuff.mark_of_the_crane.remains,if=combo_strike&cooldown.fists_of_fury.remains>3" );
+    heavy_aoe->add_action( "rushing_jade_wind,if=!buff.rushing_jade_wind.up" );
+    heavy_aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.stack=2&talent.shadowboxing_treads" );
+    heavy_aoe->add_action( "whirling_dragon_punch" );
+    heavy_aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.stack=3" );
+    heavy_aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.up&talent.shadowboxing_treads" );
+    heavy_aoe->add_action( "strike_of_the_windlord" );
+    heavy_aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=combo_strike" );
+    heavy_aoe->add_action( "spinning_crane_kick,if=min:debuff.mark_of_the_crane.remains,if=(combo_strike&chi>5&talent.storm_earth_and_fire|combo_strike&chi>4&talent.serenity)" );
+
+    // 4 Target priority
+    aoe->add_action( "spinning_crane_kick,if=combo_strike&buff.dance_of_chiji.up", "4 Targets" );
     aoe->add_action( "strike_of_the_windlord,if=talent.thunderfist" );
-    aoe->add_action( "whirling_dragon_punch,if=active_enemies>5" );
-    aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.up&(buff.teachings_of_the_monastery.stack=2|active_enemies<5)&talent.shadowboxing_treads" );
-    aoe->add_action( "whirling_dragon_punch" );
+    aoe->add_action( "fists_of_fury" );
     aoe->add_action( "spinning_crane_kick,if=buff.bonedust_brew.up&combo_strike" );
-    aoe->add_action( "fists_of_fury,if=active_enemies>3" );
+    aoe->add_action( "blackout_kick,target_if=min:debuff.mark_of_the_crane.remains,if=buff.teachings_of_the_monastery.stack=3&talent.shadowboxing_treads" );
+    aoe->add_action( "spinning_crane_kick,if=min:debuff.mark_of_the_crane.remains,if=combo_strike&cooldown.fists_of_fury.remains<3&buff.chi_energy.stack>10" );
```

Comment:
In several places, entries have two sets of conditions (`if`) where the first is almost certainly intended to be `target_if` instead.  See also lines 675, 692, etc.